src/NBitcoin/ProvenBlockHeader.cs

```diff
@@ -70,6 +70,9 @@ public class ProvenBlockHeader : PosBlockHeader
         /// <summary>Gets the size of the coinstake in bytes, the header must be serialized or deserialized for this property to be set.</summary>
         public long? CoinstakeSize { get; protected set; }
 
+        /// <summary>Gets the header size in bytes, the header must be serialized or deserialized for this property to be set.</summary>
+        public long HeaderSize { get; private set; }
```

Comment:
it's always 80
Such a constant already in base class: `public const int Size = 80;`